### PR TITLE
Add validation and test for borrowingLimit when cohort is nil

### PR DESCRIPTION
**Pull request by B1F030**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
When cohort is nil, borrowingLimit should be nil too.
After add a webhook [validation](https://github.com/kubernetes-sigs/kueue/pull/1525/files#diff-db61a388fb0c695508904a27558767caa29f58120a52804b6d0dc9df11e98e70), some tests failed:
[integration-test](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1385/pull-kueue-test-integration-main/1740211108343451648)
[unit](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/1385/pull-kueue-test-unit-main/1740211108284731392)
#### Which issue(s) this PR fixes:

#1496
#1385

#### Does this PR introduce a user-facing change?

```release-note
Add validation for clusterQueue: when cohort is empty, borrowingLimit must be nil.
```

### CODE REVIEW

**Comment by kerthcet** [comment]
/remove-kind cleanup
/kind feature

As we're adding validations for borrowingLimit.

File: pkg/webhooks/clusterqueue_webhook.go [hunk 1]
```diff
@@ -159,7 +160,7 @@ func validateResourceGroups(resourceGroups []kueue.ResourceGroup, path *field.Pa
 	return allErrs
 }
 
-func validateFlavorQuotas(flavorQuotas kueue.FlavorQuotas, coveredResources []corev1.ResourceName, path *field.Path) field.ErrorList {
+func validateFlavorQuotas(flavorQuotas kueue.FlavorQuotas, coveredResources []corev1.ResourceName, cq *kueue.ClusterQueue, path *field.Path) field.ErrorList {
```
Comment (kerthcet) [inline pkg/webhooks/clusterqueue_webhook.go:163]: cohort here is enough.

File: pkg/webhooks/clusterqueue_webhook.go [hunk 2]
```diff
@@ -189,3 +191,12 @@ func validateResourceQuantity(value resource.Quantity, fldPath *field.Path) fiel
 	}
 	return allErrs
 }
+
+// validateBorrowingLimit enforces that BorrowingLimit is nil when cohort is nil
```
Comment (kerthcet) [inline pkg/webhooks/clusterqueue_webhook.go:195]:
```suggestion
// validateBorrowingLimit enforces that BorrowingLimit must be nil when cohort is nil
```

File: pkg/webhooks/clusterqueue_webhook_test.go [hunk 3]
```diff
@@ -107,18 +107,30 @@ func TestValidateClusterQueue(t *testing.T) {
...
 			},
 		},
+		{
+			name: "flavor quota with borrowingLimit and nil cohort",
+			clusterQueue: testingutil.MakeClusterQueue("cluster-queue").
+				ResourceGroup(
+					*testingutil.MakeFlavorQuotas("x86").Resource("cpu", "1", "1").Obj()).
+				Obj(),
+			wantErr: field.ErrorList{
+				field.Invalid(resourceGroupsPath.Index(0).Child("flavors").Index(0).Child("resources").Index(0).Child("borrowingLimit"), "1", "must be nil when cohort is nil"),
```
Comment (kerthcet) [inline pkg/webhooks/clusterqueue_webhook_test.go:131]: use borrowingLimitErrorMsg directly here.

File: pkg/webhooks/clusterqueue_webhook.go [hunk 4]
```diff
@@ -36,7 +36,8 @@ import (
 )
 
 const (
-	isNegativeErrorMsg string = `must be greater than or equal to 0`
+	isNegativeErrorMsg     string = `must be greater than or equal to 0`
+	borrowingLimitErrorMsg string = `must be nil when cohort is nil`
```
Comment (kerthcet) [inline pkg/webhooks/clusterqueue_webhook.go:40]:
```suggestion
	borrowingLimitErrorMsg string = `must be nil when cohort is empty`
```

**Review by kerthcet — COMMENTED** [review]
/lgtm
/approve

**Comment by kerthcet** [comment]
Can you update the release note:
```
Add validation for clusterQueue: when cohort is empty, borrowingLimit must be nil.
```

**Comment by B1F030** [comment]
> Can you update the release note:
> 
> ```
> Add validation for clusterQueue: when cohort is empty, borrowingLimit must be nil.
> ```

Done.

File: pkg/webhooks/clusterqueue_webhook.go [hunk 5]
```diff
@@ -176,6 +177,7 @@ func validateFlavorQuotas(flavorQuotas kueue.FlavorQuotas, coveredResources []co
 		allErrs = append(allErrs, validateResourceQuantity(rq.NominalQuota, path.Child("nominalQuota"))...)
 		if rq.BorrowingLimit != nil {
 			allErrs = append(allErrs, validateResourceQuantity(*rq.BorrowingLimit, path.Child("borrowingLimit"))...)
+			allErrs = append(allErrs, validateBorrowingLimit(*rq.BorrowingLimit, cohort, path.Child("borrowingLimit"))...)
```
Comment (tenzen-y) [inline pkg/webhooks/clusterqueue_webhook.go:180]:
```suggestion
			borrowingLimitPath := path.Child("borrowingLimit")
			allErrs = append(allErrs, validateResourceQuantity(*rq.BorrowingLimit, borrowingLimitPath)...)
			allErrs = append(allErrs, validateBorrowingLimit(*rq.BorrowingLimit, cohort, borrowingLimitPath)...)
```

nit

**Review by tenzen-y — COMMENTED** [review]
Otherwise lgtm

File: pkg/webhooks/clusterqueue_webhook.go (hunk 5 above)
Comment (B1F030) [reply pkg/webhooks/clusterqueue_webhook.go]: Done.

**Review by tenzen-y — COMMENTED** [review]
Thanks!
/lgtm
/approve

**Pull request merged**